src/Core/Repositories/ICipherRepository.cs

```diff
@@ -23,7 +23,8 @@ public interface ICipherRepository : IRepository<Cipher, Guid>
         Task UpdatePartialAsync(Guid id, Guid userId, Guid? folderId, bool favorite);
         Task UpdateAttachmentAsync(CipherAttachment attachment);
         Task DeleteAttachmentAsync(Guid cipherId, string attachmentId);
-        Task DeleteAsync(IEnumerable<Guid> ids, Guid userId);
+        Task DeleteAsync(Cipher obj, bool permanent = false);
+        Task DeleteAsync(IEnumerable<Guid> ids, Guid userId, bool permanent = false);
```

Comment:
For now, let's keep the same functionality by default and set `bool permanent = true`